core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -19,162 +19,256 @@
+        $scope.activeThread.status : null;
+      $scope.tmpMessage.text = '';
+    };
+
+    $scope.clearActiveThread = function() {
+      $scope.activeThread = null;
+      _resetTmpMessageFields();
+    };
+
+    $scope.showCreateThreadModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorFeedbackCreateThread',
+        backdrop: true,
+        resolve: {},
+        controller: ['$scope', '$modalInstance', function(
+          $scope, $modalInstance) {
+          $scope.newThreadSubject = '';
+          $scope.newThreadText = '';
+
+          $scope.create = function(newThreadSubject, newThreadText) {
+            if (!newThreadSubject) {
+              warningsData.addWarning('Please specify a thread subject.');
+              return;
+            }
+            if (!newThreadText) {
+              warningsData.addWarning('Please specify a message.');
+              return;
+            }
+
+            $modalInstance.close({
+              newThreadSubject: newThreadSubject,
+              newThreadText: newThreadText
+            });
+          };
+
+          $scope.cancel = function() {
+            $modalInstance.dismiss('cancel');
+          };
+        }]
+      }).result.then(function(result) {
+        threadDataService.createNewThread(
+          result.newThreadSubject, result.newThreadText, function() {
+            $scope.clearActiveThread();
           });
-        };
+      });
+    };
+
+    var _isSuggestionOpen = function() {
+      return $scope.activeThread.status === 'open';
+    };
 
-        $scope.rejectSuggestion = function() {
-          $modalInstance.close({
-            action: 'reject'
+    var _isSuggestionValid = function() {
+      return explorationStatesService.getStates()[
+        $scope.activeThread.suggestion.state_name] !== undefined;
+    };
+
+    var _hasUnsavedChanges = function() {
+      return (changeListService.getChangeList().length > 0);
+    };
+
+    $scope.viewSuggestionBtnType = function() {
+      return (_isSuggestionOpen() && _isSuggestionValid() &&
+              !_hasUnsavedChanges() ? 'primary' : 'default');
+    };
+
+    // TODO(Allan): Implement ability to edit suggestions before applying.
+    $scope.showSuggestionModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorViewSuggestion',
+        backdrop: true,
+        size: 'lg',
+        resolve: {
+          suggestionIsOpen: function() {
+            return _isSuggestionOpen();
+          },
+          suggestionIsValid: function() {
+            return _isSuggestionValid();
+          },
+          unsavedChangesExist: function() {
+            return _hasUnsavedChanges();
+          },
+          suggestionStatus: function() {
+            return $scope.activeThread.status;
+          },
+          description: function() {
+            return $scope.activeThread.suggestion.description;
+          },
+          oldContent: function() {
```

Comment:
Interesting -- thinking about it, this is more like currentContent, right? (E.g. if the modal is opened after the suggestion has been accepted.)